include/unifex/connect_awaitable.hpp

```diff
@@ -182,31 +176,17 @@ namespace _await_cpo {
         // for the receiver to destroy the coroutine.
         co_yield [&](result_type&& result) {
               return [&] {
-                constexpr size_t valueOverloadCount =
-                    sender_value_types_t<Awaitable, count_types, single_value_type>::value;
-                static_assert(valueOverloadCount <= 1);
-
-                if constexpr (valueOverloadCount == 1) {
-                  constexpr size_t valueCount =
-                      sender_value_types_t<Awaitable, type_identity_t, count_types>::value;
-                  if constexpr (valueCount == 0) {
-                    unifex::set_value(std::move(receiver));
-                  } else if constexpr (valueCount == 1) {
-                    unifex::set_value(std::move(receiver), static_cast<result_type&&>(result));
-                  } else {
-                    std::apply(set_value_applicator<Receiver>{receiver}, (result_type&&)result);
-                  }
+                if constexpr (std::is_void_v<await_result_t<awaitable_type>>) {
+                  unifex::set_value(std::move(receiver));
                 } else {
-                  // Shouldn't complete with a value if there are no value_types
-                  // specified.
-                  std::terminate();
+                  unifex::set_value(std::move(receiver), static_cast<result_type&&>(result));
```

Comment:
I'm kind of partial to the idea that we should splat `tuple`-like things into separate arguments here.  Dunno if it can be done generically, though.  I suppose you might be able to query for whether `std::tuple_size<result_type>` works and delegate to `std::apply` if so; the docs suggest it'd work for `std::pair`, `std::tuple`, `std::array`, and any type that specializes `tuple_size` and `get`.

If this idea isn't horrible, it can come later.

Reply:
That wouldn't be appropriate in this PR.